statsmodels/stats/outliers_influence.py

```diff
@@ -186,7 +187,11 @@ def variance_inflation_factor(exog, exog_idx):
     https://en.wikipedia.org/wiki/Variance_inflation_factor
     """
     k_vars = exog.shape[1]
-    x_i = exog[:, exog_idx]
+    x_i = None
+    if isinstance(exog, pd.DataFrame):
```

Comment:
Instead of this check, could you add `np.asarray` here as you've done in the test? That way, a DataFrame and an array will pass through.